pkg/agent/svid/rotator.go

```diff
@@ -137,6 +138,25 @@ func (r *rotator) SetRotationFinishedHook(f func()) {
 	r.rotationFinishedHook = f
 }
 
+func (r *rotator) Reattest(ctx context.Context) (err error) {
+	state, ok := r.state.Value().(State)
+	if !ok {
+		return fmt.Errorf("unexpected value type: %T", r.state.Value())
+	}
+
+	if state.Reattestable && !r.c.DisableReattestToRenew {
```

Comment:
I would probably separate this in two different conditions and return separate errors so it's easier to understand and debug what happened.